src/util/request-manager.js

```diff
@@ -407,10 +413,10 @@ export default class RequestManager {
 
     let proxy = this.httpProxy;
     if (params.url.startsWith('https:')) {
-      proxy = this.httpsProxy || proxy;
+      proxy = this.httpsProxy;
     }
-    if (proxy) {
-      params.proxy = proxy;
+    if (proxy && typeof proxy === 'string') {
+      params.proxy = String(proxy);
```

Comment:
So we check the type and then cast it to a `String` again? I'm guessing this is due to some messed up Flow typing. Shall we fix it since we are here?

Reply:
Yes this was purely to make Flow happy, because `proxy` is `string | boolean` and it won't let you assign it to `parame.proxy` because that only takes `string`. However I don't think the `typeof` check was actually needed.